src/Microsoft.AspNet.Mvc.Core/HttpResponseStreamWriter.cs

```diff
@@ -42,6 +44,11 @@ public HttpResponseStreamWriter(Stream stream, Encoding encoding, int bufferSize
                 throw new ArgumentNullException(nameof(stream));
             }
 
+            if (!stream.CanWrite)
```

Comment:
It would, but there might be a time gap between creating the writer and writing.